source/ScreensaverHero.hx

```diff
@@ -0,0 +1,92 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.math.FlxRandom;
+import flixel.util.FlxSpriteUtil;
+
+class ScreensaverHero extends FlxSprite
+{
+	public var direction:Int;
+
+	public function new(x:Int, y:Int)
+	{
+		super(x, y);
+		var randomDirection:FlxRandom = new FlxRandom();
+		this.direction = randomDirection.int(0, 7);
+	}
+
+	override function update(elapsed:Float)
+	{
+		if (this.x < 500)
+		{
+			this.color = 0xff0000;
+		}
+		if (this.x > 500)
+		{
+			this.color = 0x00ff00;
+		}
+		if (this.direction == 0)
+		{
+			this.x--;
+		}
+		else if (this.direction == 1)
+		{
+			this.x--;
+			this.y--;
+		}
+		else if (this.direction == 2)
+		{
+			this.y--;
+		}
+		else if (this.direction == 3)
+		{
+			this.x++;
+		}
+		else if (this.direction == 4)
+		{
+			this.x++;
+			this.y++;
+		}
+		else if (this.direction == 5)
+		{
+			this.y++;
+		}
+		else if (this.direction == 6)
+		{
+			this.x--;
+			this.y++;
+		}
+		else
+		{
+			this.x++;
+			this.y--;
+		}
```

Comment:
This ends up accomplishing the goal of our assignment, but it's extremely maintenance-unfriendly, especially when it's put in the middle of a method like `update()` that's already likely to be a complex pile of unrelated behaviors.

At a minimum, let's move your direction-picking code to its own private method. I think it'd be best if we could get rid of this big conditional block entirely.  

As an example: you could generate two numbers in the constructor, one for X movement and one for Y movement, each with integer values from -1 to 1, then just add them to your current x and y each update.